`useless-super-delegation` when overriding `Exception` to have different default argument

### Bug description

```python
# pylint: disable=missing-module-docstring,missing-class-docstring
class CustomError(Exception):
    def __init__(self, message="default"):
        super().__init__(message)
```

### Configuration

_No response_

### Command used

```shell
pylint a.py
```


### Pylint output

```shell
************* Module a
a.py:3:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)
```


### Expected behavior

No error, because this is not a useless super delegation. The method signature is different. A default value has been introduced. (Also discussed in #839 from which the rule was introduced).

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.9.13 (main, Aug 11 2022, 13:40:36)
[Clang 13.1.6 (clang-1316.0.21.2.5)]
```


### OS / Environment

macOS 12.4

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/7319.bugfix
+++ b/doc/whatsnew/fragments/7319.bugfix
@@ -1,0 +1,4 @@
+Prevent `useless-parent-delegation` for delegating to a builtin
+written in C (e.g. `Exception.__init__`) with non-self arguments.
+
+Closes #7319

--- a/pylint/checkers/classes/class_checker.py
+++ b/pylint/checkers/classes/class_checker.py
@@ -1290,6 +1290,8 @@ a metaclass class method.",
                 or _has_different_parameters_default_value(
                     meth_node.args, function.args
                 )
+                # arguments to builtins such as Exception.__init__() cannot be inspected
+                or (meth_node.args.args is None and function.argnames() != ["self"])
             ):
                 return
             break

--- a/tests/functional/u/useless/useless_parent_delegation.py
+++ b/tests/functional/u/useless/useless_parent_delegation.py
@@ -425,3 +425,8 @@ class Fruit:
 class Lemon(Fruit):
     def __init__(*, tastes_bitter=True):
         super().__init__(tastes_bitter=tastes_bitter)
+
+
+class CustomError(Exception):
+    def __init__(self, message="default"):
+        super().__init__(message)

